server/player/player.go

```diff
@@ -545,9 +547,46 @@ func (p *Player) Hurt(dmg float64, source damage.Source) (float64, bool) {
 		return 0, true
 	}
 
+	w, pos := p.World(), p.Position()
 	totalDamage := p.FinalDamageFrom(dmg, source)
-	damageLeft := totalDamage
+	if ok, _ := p.Blocking(); ok && source.ReducedByArmour() {
+		affected := true
+		if src, ok := source.(damage.SourceEntityAttack); ok {
+			diff := p.Position().Sub(src.Attacker.Position())
+			diff[1] = 0
+			if diff.Dot(entity.DirectionVector(p)) >= 0.0 {
+				affected = false
+			}
+		}
+		if affected {
+			w.PlaySound(pos, sound.ShieldBlock{})
+			p.SetBlockingDelay(time.Millisecond * 250)
+
+			if src, ok := source.(damage.SourceEntityAttack); ok {
+				if l, ok := src.Attacker.(entity.Living); ok {
+					l.KnockBack(pos, 0.5, 0.4)
+				}
+				if a, ok := src.Attacker.(*Player); ok {
+					held, _ := a.HeldItems()
+					if _, ok := held.Item().(item.Axe); ok {
+						p.SetBlockingDelay(time.Second * 5)
+					}
+				}
+			}
+			if dmg >= 3.0 {
+				i := int(math.Ceil(totalDamage))
+				held, other := p.HeldItems()
+				if _, ok := held.Item().(item.Shield); ok {
+					p.SetHeldItems(p.damageItem(held, i), other)
+				} else {
+					p.SetHeldItems(held, p.damageItem(other, i))
+				}
+			}
+			return 0, false
+		}
+	}
```

Comment:
Could you pull this out in its own method and add some documentation (if and where relevant)? This is a massive block of code and Player.Hurt() is already a pretty big function as it is.

Reply:
any method name suggestions?

Reply:
this has been resolved, once the conflicts are resolved, are we ready to merge?